prow/gerrit/reporter/reporter.go

```diff
@@ -205,3 +228,65 @@ func (c *Client) Report(pj *v1.ProwJob) ([]*v1.ProwJob, error) {
 
 	return toReportJobs, nil
 }
+
+func statusIcon(state v1.ProwJobState) string {
+	icon, ok := stateIcon[state]
+	if !ok {
+		return prohibited
+	}
+	return icon
+}
+
+func jobFromPJ(pj *v1.ProwJob) *Job {
+	return &Job{Name: pj.Spec.Job, State: string(pj.Status.State), icon: statusIcon(pj.Status.State), url: pj.Status.URL}
+}
+
+func (j *Job) serialize() string {
+	return fmt.Sprintf(jobReportFormat, j.icon, j.Name, strings.ToUpper(j.State), j.url)
+}
+
+func deserializeJob(s string) *Job {
+	j := &Job{}
+	n, err := fmt.Sscanf(s, jobReportFormat, &j.icon, &j.Name, &j.State, &j.url)
+	if err != nil || n != 4 {
+		logrus.Debugf("Could not deserialize %s to a job: %v", s, err)
+		return nil
+	}
+	return j
+}
+
+func generateReport(pjs []*v1.ProwJob) *JobReport {
+	report := &JobReport{total: len(pjs)}
+	for _, pj := range pjs {
+		job := jobFromPJ(pj)
+		report.Jobs = append(report.Jobs, job)
+		if pj.Status.State == v1.SuccessState {
+			report.success++
+		}
+
+		report.message += job.serialize()
+		report.message += "\n"
+
+	}
+	report.header = defaultProwHeader
+	report.header += fmt.Sprintf(" %d out of %d pjs passed!\n", report.success, report.total)
+	return report
+}
+
+// ParseReport creates a jobReport from a string, nil if cannot parse
+func ParseReport(message string) *JobReport {
```

Comment:
would like to see some test cases for these parsing functions

Reply:
done